small Enum fix with instanceof check; YUI-style docs for ENum

mattdesl opened the pull request.

mattdesl committed:
adding YUI doc for Enum; ensuring instanceof check is accurate for no-arg enum symbols;
also, the returned Enum object is no longer frozen. this is because in practice many debs will want to add their own "static" constants/etc to the Enum.

mattdesl committed:
adding YUI doc for Enum; ensuring instanceof check is accurate for no-arg enum symbols

mattdesl committed:
Merge branch 'enum-fix' of https://github.com/mattdesl/jsOOP into enum-fix

mikkoh commented:
Maybe just thinking out loud here. But perhaps the second property of `Enum` should be a `new Class({});` perhaps it should just be an descriptor object. This way you could inject the base class automatically. Less for people to worry about. You can check yourself if it `Extends` something. If the descriptor does extend something other than `Enum.Base` you could throw an error.

I think this would perhaps make the syntax a bit more consistent with Class. Feels less like an "extension" on top of Class.

mattdesl committed:
moving source to js/jsOOP to reflect new base class.
adding Enum to tests
adding optional shorthand for Enum, no need to require Class anymore.

mattdesl committed:
updating enum docs
